App/Composition/SmiliePicker/SmilieSearchViewModel.swift

```diff
@@ -196,11 +196,18 @@ final class SmilieSearchViewModel: ObservableObject {
     }
     
     func updateLastUsedDate(for smilie: Smilie) {
-        guard let context = dataStore.managedObjectContext else { return }
+        guard let context = dataStore.managedObjectContext,
+              let smilieText = smilie.text else { return }
```

Comment:
see it's funny because this is exactly wrong: we should not be pawing at managed properties outside a context.perform[AndWait]. is there any way to avoid the compiler error while only accessing `smilie.text` within the `context.perform` block?